src/solvers/dgsem_tree/indicators_2d.jl

```diff
@@ -24,6 +24,100 @@ function create_cache(typ::Type{IndicatorHennemannGassner}, mesh, equations::Abs
   create_cache(typ, equations, dg.basis)
 end
 
+# Overload indicator when ShallowWaterEquations2D is used to apply full FV method on cells
+# containing some "dry" LGL nodes. That is, if an element is partially "wet" then it becomes a
+# full FV element.
```

Comment:
This should really be another indicator since it comes with quite a surprise. It could either be a new standalone indicator or a general structure wrapping an indicator and setting more elements as pure FV depending on some conditions.

Reply:
It could probably be a new indicator to avoid confusion. This opens up the issue of giving it a good descriptive name.

Reply:
If the primary concern is flexibility instead of performance, you could create an `IndicatorDry` (or something similar) that wraps a given `indicator` such as `IndicatorHennemannGassner`. When called, it calls the underlying `indicator` first and then gos through all elements again, flagging dry elements to use pure FV stuff, too.

Reply:
@andrewwinters5000 @ranocha @sloede I created a new indicator by copying the structure of the original `IndicatorHennemannGassner`. I am not completely sure if I did it in the way you want it to be. Maybe you can comment on that

Reply:
I'm not sure whether it's a good coding approach to duplicate such an amout of code. Did you run into problems with the approach of 

> a general structure wrapping an indicator and setting more elements as pure FV depending on some conditions

?

Reply:
From my perspective I think it is okay to have this specialized indicator with some code duplication. My reasoning is that, at the moment, these two indicators will live in Trixi `main`, but soon they will be separated into different packages. So it would be helpful for those using the (yet to be created) TrixiShallowWater package to see a full picture of how the indicator is computed in the specific shallow water case.

Reply:
Ok